vms/proposervm/state_syncable_vm_test.go

```diff
@@ -52,7 +53,10 @@ func helperBuildStateSyncTestObjects(t *testing.T) (*fullVM, *VM) {
 	innerVM.LastAcceptedF = snowmantest.MakeLastAcceptedBlockF(
 		[]*snowmantest.Block{snowmantest.Genesis},
 	)
-	innerVM.GetBlockF = func(context.Context, ids.ID) (snowman.Block, error) {
+	innerVM.GetBlockF = func(_ context.Context, blkID ids.ID) (snowman.Block, error) {
+		if blkID != snowmantest.Genesis.ID() {
+			return nil, database.ErrNotFound
+		}
```

Comment:
Ensure that the default test setup does not silently ignore the `blkID` parameter and return the genesis